tests/utils/profilehandler/profile_handler.go

```diff
@@ -332,11 +344,97 @@ func GenerateClusterCreateFlags(profile *Profile, client *rosacli.Client) ([]str
+			HostPrefix:  "25",
+		}
+		flags = append(flags,
+			"--machine-cidr", networking.MachineCIDR, // Placeholder, it should be vpc CIDR
+			"--service-cidr", networking.ServiceCIDR,
+			"--pod-cidr", networking.PodCIDR,
+			"--host-prefix", networking.HostPrefix,
+		)
+		clusterConfiguration.Networking = networking
+	}
 	if profile.ClusterConfig.BYOVPC {
-		PrepareSubnetsDummy("", profile.Region, "")
+		vpcPrefix := common.TrimNameByLength(clusterName, 20)
+		log.Logger.Info("Got BYOVPC set to true. Going to prepare subnets")
+		cidrValue := con.DefaultVPCCIDRValue
+		if profile.ClusterConfig.NetWorkingSet {
+			cidrValue = clusterConfiguration.Networking.MachineCIDR
+		}
+		vpc, err := PrepareVPC(profile.Region, vpcPrefix, cidrValue)
+		if err != nil {
+			return flags, err
+		}
+		userData.VpcID = vpc.VpcID
+		zones := strings.Split(profile.ClusterConfig.Zones, ",")
+		zones = common.RemoveFromStringSlice(zones, "")
+		subnets, err := PrepareSubnets(vpc, profile.Region, zones, profile.ClusterConfig.MultiAZ)
+		if err != nil {
+			return flags, err
+		}
+		subnetsFlagValue := strings.Join(append(subnets["private"], subnets["public"]...), ",")
+		clusterConfiguration.Subnets = &ClusterConfigure.Subnets{
+			PrivateSubnetIds: strings.Join(subnets["private"], ","),
+			PublicSubnetIds:  strings.Join(subnets["public"], ","),
+		}
+		if profile.ClusterConfig.PrivateLink {
+			log.Logger.Info("Got private link set to true. Only set private subnets to cluster flags")
+			subnetsFlagValue = strings.Join(subnets["private"], ",")
+			clusterConfiguration.Subnets = &ClusterConfigure.Subnets{
+				PrivateSubnetIds: strings.Join(subnets["private"], ","),
+			}
+		}
+		flags = append(flags,
+			"--subnet-ids", subnetsFlagValue)
+
+		if profile.ClusterConfig.AdditionalSGNumber != 0 {
+			securityGroups, err := PrepareAdditionalSecurityGroups(vpc, profile.ClusterConfig.AdditionalSGNumber, vpcPrefix)
+			if err != nil {
+				return flags, err
+			}
+			computeSGs := strings.Join(securityGroups, ",")
+			infraSGs := strings.Join(securityGroups, ",")
+			controlPlaneSGs := strings.Join(securityGroups, ",")
+			flags = append(flags,
+				"--additional-infra-security-group-ids", infraSGs,
+				"--additional-control-plane-security-group-ids", controlPlaneSGs,
+				"--additional-compute-security-group-ids", computeSGs,
+			)
+			clusterConfiguration.AdditionalSecurityGroups = &ClusterConfigure.AdditionalSecurityGroups{
+				ControlPlaneSecurityGroups: controlPlaneSGs,
+				InfraSecurityGroups:        infraSGs,
+				WorkerSecurityGroups:       computeSGs,
+			}
+		}
+		if profile.ClusterConfig.ProxyEnabled {
+			proxy, err := PrepareProxy(vpc, profile.Region, config.Test.ProxySSHPemFile, config.Test.ProxyCABundleFile)
+			if err != nil {
+				return flags, err
+			}
+
+			clusterConfiguration.Proxy = &ClusterConfigure.Proxy{
+				Enabled:         profile.ClusterConfig.ProxyEnabled,
+				Http:            proxy.HTTPProxy,
+				Https:           proxy.HTTPsProxy,
```

Comment:
I think it is better to keep the uppercase and lowercase uniform, `Http`  and `HTTPProxy`  -> `HttpProxy`

Reply:
No. HTTP is shortage of `Hypertext Transfer Protocol`. I will keep HTTP since it's more readable.